Analytics/Request/BlockingRequestHandler.cs

```diff
@@ -225,19 +223,32 @@ public async Task MakeRequest(Batch batch)
                     if (_client.Config.CompressRequest)
                         content.Headers.ContentEncoding.Add("gzip");
 
-                    var response = await _httpClient.PostAsync(uri, content).ConfigureAwait(false);
+                    HttpResponseMessage response = null;
+                    bool retry = false;
+                    try
+                    {
+                        response = await _httpClient.PostAsync(uri, content).ConfigureAwait(false);
+                    }
+                    catch (System.Threading.Tasks.TaskCanceledException)
+                    {
+                        retry = true;
+                    }
+                    catch (HttpRequestException)
+                    {
+                        retry = true;
+                    }
 
                     watch.Stop();
 
-                    if (response.StatusCode == HttpStatusCode.OK)
+                    if (response != null && response.StatusCode == HttpStatusCode.OK)
                     {
                         Succeed(batch, watch.ElapsedMilliseconds);
                         break;
                     }
                     else
                     {
-                        statusCode = (int)response.StatusCode;
-                        if ((statusCode >= 500 && statusCode <= 600) || statusCode == 429)
+                        statusCode = response != null ? (int)response.StatusCode : 0;
+                        if ((statusCode >= 500 && statusCode <= 600) || statusCode == 429 || retry)
```

Comment:
I think the check should be for `[500, 600)`, i.e. `< 600` instead of `<= 600`.

Reply:
Hi, @f2prateek 
I changed the status code range for retry.